hazelcast-jet-core/src/test/java/com/hazelcast/jet/config/YamlJetConfigBuilderConfigResolutionTest.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.config;
+
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.jet.impl.config.YamlJetConfigBuilder;
+import com.hazelcast.jet.test.JetDeclarativeConfigFileHelper;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+
+import static com.hazelcast.internal.config.DeclarativeConfigUtil.YAML_ACCEPTED_SUFFIXES_STRING;
+import static com.hazelcast.jet.impl.config.JetDeclarativeConfigUtil.SYSPROP_JET_CONFIG;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastSerialClassRunner.class)
+public class YamlJetConfigBuilderConfigResolutionTest {
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    private JetDeclarativeConfigFileHelper helper = new JetDeclarativeConfigFileHelper();
+
+    @Before
```

Comment:
Bit weird for the `tearDown()` method to be annotated with `@Before`... Mistake?

Reply:
it should clear out the system property if present, so we also run this method before each run.